arwen/crypto/cryptoei.go

```diff
@@ -37,7 +37,7 @@ func sha256(context unsafe.Pointer, dataOffset int32, length int32, resultOffset
 	crypto := arwen.GetCryptoContext(context)
 
 	data, err := runtime.MemLoad(dataOffset, length)
-	if withFault(err, context) {
+	if arwen.WithFault(err, context) {
```

Comment:
there is no need for return for these functions if we are getting out of arwen execution with error and breakpoint. These returns are for the SC developer to know it was an error, so it might have some fallback mechanism implemented there.

Reply:
Changed `WithError` to be more flexible. Now the API may (or may not) set a breakpoint, depending on the internal state of the Runtime context. If execution is allowed to continue, the return value will be useful to the SC, as discussed.